### DISCUSSION TITLE
Warp renderer with cylinder

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#139] [Q&A] [answered]
I am trying to use warp renderer with a cylinder, and it isn't happy. I made it in omniverse (create-mesh-cylinder).  The error is:

```
lib\site-packages\warp\render\render_usd.py", line 528, in render_mesh
    idxs = np.array(indices).reshape(-1, 3)
ValueError: cannot reshape array of size 320 into shape (3)
```

Is there some step I am missing?  

I am using this example https://github.com/NVIDIA/warp/blob/main/examples/example_mesh.py

**Comment by mmacklin** [comment, marked as answer]
Hi @cadop , I am guessing it's because the cylinder is a quad mesh, but `render_mesh` assumes a triangle mesh.

You would need to triangulate it first, take a look at the code here on how to do it: https://github.com/NVIDIA/warp/blob/main/warp/tests/test_mesh_query_point.py#L248

Cheers,
Miles

**Reply by cadop** [reply]
Thanks! solved it

**Reply by cadop** [reply]
What do you think about adding this as a built-in function? Maybe here:

https://github.com/NVIDIA/warp/blob/1535cdecc626c9f527b99467ed94dbf8a20cc00a/warp/sim/model.py#L135

And have a default flag with `triangles = True`, but if its false, runs the triangulate method.
